xclim/core/units.py

```diff
@@ -57,7 +57,7 @@
 units = deepcopy(cf_xarray.units.units)
 # Changing the default string format for units/quantities. cf is implemented by cf-xarray
 # g is the most versatile float format.
-units.default_format = "gcf"
+units.formatter.default_format = "gcf"
```

Comment:
This will require pinning pint (which will implicitly pin numpy > 2)!

If this is not ok to you, we could to a if-else with the version?

Reply:
Sounds good! I'll adapt that.